storage-proofs/src/circuit/zigzag.rs

```diff
@@ -84,6 +85,11 @@ impl<'a, H: Hasher> Circuit<Bls12> for ZigZagCircuit<'a, Bls12, H> {
         let graph = &self.public_params.graph;
         let layer_challenges = &self.public_params.layer_challenges;
 
+        if !cfg!(feature = "allow-small-degree") {
+            assert_eq!(graph.base_graph().degree(), BASE_DEGREE);
```

Comment:
we should validate this in the base graph as well

Reply:
@dignifiedquire ok, I added degree validation to `ZigZagBucketGraph`'s and `BucketGraph`'s constructors and changed all the places in the tests where `BucketGraph` was being instantiated with a degree not equal to `BASE_DEGREE`.